### ISSUE TITLE
Update to latest version of publish-plugin

### ISSUE THREAD

**Issue by TimvdLippe** [shipkit/shipkit-changelog#146]
https://github.com/gradle-nexus/publish-plugin is the new location for the publish plugin. It fixes some issues that we consistently encounter on Mockito. For example, the 5.9.0 release is stuck on https://github.com/mockito/mockito/actions/runs/7519868803/job/20471236051 with

```
> Staging repository is not in desired state [released, not_found]: StagingRepository(id=orgmockito-1112, state=closed, transitioning=false). It is unexpected. Please check the Nexus logs using its web interface - it can be caused by validation rules violation (e.g. publishing artifacts with the same version again). If not, please report it to https://github.com/gradle-nexus/publish-plugin/issues/ with the '--info' logs.
```

Which is fixed by https://github.com/gradle-nexus/publish-plugin/issues/52 and in the latest release.

**Comment by TimvdLippe** [comment]
This would be fixed by merging https://github.com/shipkit/shipkit-changelog/pull/137

**Comment by mockitoguy** [comment]
Just merged #137

**Comment by TimvdLippe** [comment]
Thanks! Can you also publish a release for it?
